subsys/net/l2/ppp/ppp_l2.c

```diff
@@ -284,11 +284,18 @@ static void carrier_on_off(struct k_work *work)
 		ppp_mgmt_raise_carrier_on_event(ctx->iface);
 		net_if_up(ctx->iface);
 	} else {
-		ppp_lower_down(ctx);
-		ppp_change_phase(ctx, PPP_DEAD);
+		if (ppp_lcp) {
+			ppp_lcp->close(ctx, "Shutdown");
+			/* signaling for the carrier off event is done from the LCP callback */
+		} else {
+			if (ppp_lcp) {
```

Comment:
I think it will never enter this? If `ppp_lcp` exists it would enter the first `if` @ L287

Reply:
The previous path took the LCP state machine down, without sending an actual terminate to the modem. This left the LCP state machine in the opened state at the modem side. In my case (without gsm_mux) this prevented it from answering to AT commands after closing down the connection.
In the new situation, the lcp_close function is called instead of the lcp_down, which sends out a TERMINATE_REQ to the modem, which it has to respond to with an TERMINATE_ACK. As the code shown here is called from a workqueue, we cannot wait for the TERMINATE_ACK to arrive and the final closing statements have been moved to the used callback function lcp_finished.
In this new situation the statements ppp_mgmt_raise_carrier_off_event(ctx->iface); and net_if_carrier_down(ctx->iface); are just delayed until the LCP connection has properly been terminated. Calling net_if_carrier_down earlier would prevent sending out LCP messages and properly terminating.

Reply:
Reading it I see your point, will fix it